app/controllers/users_controller.rb

```diff
@@ -0,0 +1,38 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+@user = User.new(user_params)    # Not the final implementation!
+    if @user.save
+    flash[:success] = "Welcome to the Private Events website!"
+    log_in(@user)
+    redirect_to @user
+
+      # Handle a successful save.
+    else
+      render 'new'
+    end
+  end
+
+  def show
+    @user = User.find(params[:id])
+   
```

Comment:
why find the user? we can use the current user and the relationship to events to get all that.
let's have:
`@created_events = current_user.events`
`@past_events = current_user.events.previous`
`@upcoming_events = current_user.events.upcoming`

Reply:
Added the previous and upcoming events under created and attending events of the user.